CONTRIBUTING.md

```diff
@@ -33,29 +33,35 @@ cargo clippy --workspace --tests -- --deny warnings
-
 ## Contribution Tips
 
 - If you've never made a pull request on github before, [this](https://www.freecodecamp.org/news/how-to-make-your-first-pull-request-on-github-3/) will be a good place to start.
-- Create an issue if the purpose of a struct/field/type/function/... is not immediately clear from its name or nearby comments.
 - You can find good first issues [here][good-first-issues]. Once you have gained some experience you can take a look at the [intermediate issues](https://github.com/roc-lang/roc/issues?q=is%3Aopen+is%3Aissue+label%3A%22intermediate+issue%22).
-- [Fork](https://github.com/roc-lang/roc/fork) the repo so that you can apply your changes first on your own copy of the roc repo.
-- It's a good idea to open a draft pull request as you begin working on something. This way, others can see that you're working on it, which avoids duplicate effort, and others can give feedback sooner rather than later if they notice a problem in the direction things are going. Click the button "ready for review" when it's ready.
+- It's a good idea to open a draft pull request as you begin working on something. This way, others can see that you're working on it, which avoids duplicate effort, and others can give important feedback sooner rather than later. Click the button "ready for review" when it's ready.
+- The [compiler's README](https://github.com/roc-lang/roc/tree/main/crates/compiler) contains important info.
+- The AI chat in the [cursor editor](https://www.cursor.com/) can also help you find your way in the codebase.
+
+<details>
+<summary>:beetle: Debugging Tips</summary>
+
+- At the bottom of [.cargo/config.toml](https://github.com/roc-lang/roc/blob/main/.cargo/config.toml) we have useful debug flags that activate certain debug prints.
+- For Roc code; minimize the code that produces the issue.
+- For segmentation faults:
+    + In general we recommend using linux to investigate, it has better tools for this. 
+    + Use `roc build myApp.roc --linker=legacy` followed by `valgrind ./myApp`.
+    + Use gdb to step through the code, [this gdb script](https://roc.zulipchat.com/#narrow/stream/395097-compiler-development/topic/gdb.20script/near/424422545) can be helpful.
+    + Inspect the generated LLVM IR (`roc build myApp.roc --emit-llvm-ir`) between Roc code that encounters the segfault and code that doesn't.
+  
+
+</details>
 
 ### Commit signing
 
-All your commits need to be signed [to prevent impersonation](https://dev.to/martiliones/how-i-got-linus-torvalds-in-my-contributors-on-github-3k4g). Check out [our guide for commit signing](devtools/signing.md).
+All your commits need to be signed [to prevent impersonation](https://dev.to/martiliones/how-i-got-linus-torvalds-in-my-contributors-on-github-3k4g).  
+Check out [our guide for commit signing](devtools/signing.md).
 
-#### Commit signing on NixOS
+<details>
+<summary>⚠️ Tip for NixOS users</summary>
```

Comment:
This is only relevant for a small portion of readers, so going with a collapsible section seems best.